### PR TITLE
chore: update used jazzer version to 0.16.0

### PR DESCRIPTION
**Pull request by jochil**
### Motivation & Context
We want to make sure that users can immediately use jazzer with the newest features

### Description
Update everything to jazzer 0.16.0

@MarkusZoppelt I noticed that we use jazzer 0.13.3 in our vhs showcase (in the readme)... but I thought it is not urgent to update this as well. WDYT?

### CODE REVIEW

**Comment by MarkusZoppelt** [comment]
> @MarkusZoppelt I noticed that we use jazzer 0.13.3 in our vhs showcase (in the readme)... but I thought it is not urgent to update this as well. WDYT?

@jochil I think we can update the contents of the `.tape` file(s) as well. If we want to update the gif itself, we will already have the latest version handy and don't need to remember to update it then.

**Review by MarkusZoppelt — APPROVED** [review]

**Review by florianGla — APPROVED** [review]

**Comment by jochil** [comment]
@fmeum there were a few changes necessary to make our tests work with the new jazzer version. Would be great if you can check if they make sense (last commit in this PR) according to the actual jazzer changes

File: integration-tests/java-maven/java_maven_test.go [hunk 1]
```diff
@@ -127,7 +127,7 @@ func TestIntegration_Maven_InitCreateRun(t *testing.T) {
 	// When minijail is used, the artifact prefix is set to the minijail
 	// output path
 	cifuzzRunner.Run(t, &shared.RunOptions{
-		ExpectedOutputs: []*regexp.Regexp{regexp.MustCompile(`artifact_prefix='./'`)},
+		ExpectedOutputs: []*regexp.Regexp{regexp.MustCompile(`artifact_prefix=`)},
```
[the remark this answers is not in the document]
Comment (jochil) [reply integration-tests/java-maven/java_maven_test.go:130]: I was confused by this one too. This should be the relevant output:
```
...
== Java Exception: com.code_intelligence.jazzer.api.FuzzerSecurityIssueHigh: Remote Code Execution
Unrestricted class/object creation based on externally controlled data may allow
remote code execution depending on available classes on the classpath.
	at jaz.Zer.reportFinding(Zer.java:100)
	at jaz.Zer.reportFindingIfEnabled(Zer.java:95)
	at jaz.Zer.<init>(Zer.java:68)
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:67)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at java.base/java.lang.reflect.ReflectAccess.newInstance(ReflectAccess.java:128)
	at java.base/jdk.internal.reflect.ReflectionFactory.newInstance(ReflectionFactory.java:341)
	at java.base/java.lang.Class.newInstance(Class.java:677)
	at com.example.ExploreMe.exploreMe(ExploreMe.java:19)
	at com.example.FuzzTestCase.myFuzzTest(FuzzTestCase.java:20)
DEDUP_TOKEN: 31c5fae395ad0ea8
== libFuzzer crashing input ==
MS: 0 ; base unit: 0000000000000000000000000000000000000000
0x40,0x6a,0x61,0x7a,0x2e,0x5a,0x65,0x72,0xff,0x40,0xd,0xa6,0x0,0x2,0xff,0xff,0xff,0x50,
@jaz.Zer\377@\015\246\000\002\377\377\377P
artifact_prefix='/tmp/cifuzz-maven-testdata-611582972/src/test/resources/com/example/FuzzTestCaseInputs/'; Test unit written to /tmp/cifuzz-maven-testdata-611582972/src/test/resources/com/example/FuzzTestCaseInputs/crash-ff625e8e69066e2ea2f70b8fd61bdcc9f426ec0d
Base64: QGphei5aZXL/QA2mAAL///9Q
Successfully initialized fuzzer with seed inputs

Use 'cifuzz finding <finding name>' for details on a finding.

🔍 Copied input file from /tmp/cifuzz-maven-testdata-611582972/src/test/resources/com/example/FuzzTestCaseInputs/crash-ff625e8e69066e2ea2f70b8fd61bdcc9f426ec0d to /tmp/cifuzz-maven-testdata-611582972/.cifuzz-findings/incredible_walrus/crashing-input
💥 [incredible_walrus] Security Issue: Remote Code Execution in exploreMe (com.example.ExploreMe:19)

Note: The reproducing inputs have been copied to the seed corpus at:

    src/test/resources/com/example/FuzzTestCaseInputs/incredible_walrus

They will now be used as a seed input for all runs of the fuzz test,
including remote runs with artifacts created via 'cifuzz bundle' and
regression tests. For more information on regression tests, see:

    https://github.com/CodeIntelligenceTesting/cifuzz/blob/main/docs/Regression-Testing.md

Execution time: 3s
Average exec/s: n/a
Findings:       1
Corpus entries: 6 (+18446744073709551612)
Skipping upload of findings because no project was specified and running in non-interactive mode.
    java_maven_test.go:129: 
        	Error Trace:	/home/jochen/dev/workspace/ci/cifuzz/integration-tests/shared/cifuzz_runner.go:211
        	           				/home/jochen/dev/workspace/ci/cifuzz/integration-tests/java-maven/java_maven_test.go:129
        	Error:      	Should be true
        	Test:       	TestIntegration_Maven_InitCreateRun
        	Messages:   	Did not see ["artifact_prefix='./'"] in fuzzer output
--- FAIL: TestIntegration_Maven_InitCreateRun (23.18s)
FAIL
FAIL	code-intelligence.com/cifuzz/integration-tests/java-maven	23.179s
```
it happens for example when running `go test ./integration-tests/... -run  TestIntegration_Maven_InitCreateRun  -failfast -v`

or in this pipeline: https://github.com/CodeIntelligenceTesting/cifuzz/actions/runs/4449146427/jobs/7812913044#step:4:9746

[the remark this answers is not in the document]
Comment (bertschneider) [reply integration-tests/java-maven/java_maven_test.go:130]: Was "just" the relative path changed to an absolute one or are the files actually placed into wrong folders?

[the remark this answers is not in the document]
Comment (bertschneider) [reply integration-tests/java-maven/java_maven_test.go:130]: Findings should be placed in the findings dir (`<pwd>/src/test/resources/<package>/<TestClass>Inputs`) or, if not present, the current working directory. Both paths are given as absolute paths now. This should still be the old behavior.

Are those the finding dirs you expect? I couldn't find anything about those in the tests.

[the remark this answers is not in the document]
Comment (jochil) [reply integration-tests/java-maven/java_maven_test.go:130]: I double checked locally and everything seems fine... findings are in the same place as before. It really seems like the only difference is the replacement of `./` with the actual `<pwd>`... can this be something coming from libfuzzer?

[the remark this answers is not in the document]
Comment (bertschneider) [reply integration-tests/java-maven/java_maven_test.go:130]: No, that's on us :grimacing: We consolidated the CLI and JUnit handling and missed that the logged path would be different.

[the remark this answers is not in the document]
Comment (fmeum) [reply integration-tests/java-maven/java_maven_test.go:130]: This is mostly working as intended, but there is a caveat: We use the second directory if it exists, but fall back to the first one if not. We also create the second directory if we can find the test resource directory, but Maven doesn't copy that to `target` if it is empty. I can see that happening here on the first run - the second one has the finding from the first one.

@jochil Could you still file a Jazzer issue? We should try to make this less confusing, even if Maven is involved here.
